Task-49046 : adding user have the same name as deleted one

Noussairhamrit opened the pull request:
ISSUES : when adding user have the same name as deleted one  an error message is displayed, with an unknown error message
FIX: To resolve this issue I worked on two  projects gatein-portal and social so in social  firstly i throw ObjectAlreadyExistsException instead of RuntimeException in 'SocialUserEventListenerImpl' also
I added a condition in the method 'handleError' if Error === 'USER:ALREADY_EXISTS_DELETED new error message displayed this error message is declared in 'Portlet_en.properties'

Related to https://github.com/Meeds-io/gatein-portal/pull/311.

Amine0008 committed:
task#47237: remove avatar inside comment block (#1164)

ISSUE: UX improvement, increasing the comment block size by removing the avatar image
FIX: removed the avatar image for comments and readjusted the margin styling

abenmbarek committed:
TASK-51126 : User's profile: Spaces in common: incorrect values (#1155)

ISSUES : incorrect counting of common spaces values in the rest service
FIX : add a new SQL query that returns the common spaces between two given users and use it into a new added service.
without adding unit tests

houssemexo26 committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

Amine0008 committed:
undid changes in task47237 where we removed the users avatar for the activity  stream page (#1167)

ISSUE: reverting to previous behaviour where we removed the users avatar
FIX: undid changes in task 47237

Amine0008 committed:
49308: Add Ellipsis for the blue chips on the settings page for responsive layout (#1129)

ISSUE: in the settings page we need to display the blue label chips with ellipsis for mobile web
FIX: added custom style that will be applied to these chips only when the application is opened on a mobile browser

Amine0008 committed:
changed wraper element for the v-chips components to conserve their default behaviour of resizing (#1146)

ISSUE: the previous method of applying ellipses include hard-coded width length for the label chips
FIX: changed the wrapper element for the v-chip components to preserve their default resizing behavior
Co-authored-by: Ali HAMDI <ahamdi@exoplatform.com>

Amine0008 committed:
changed wraper element for the v-chips components to conserve their default behaviour of resizing (#1146)

ISSUE: the previous method of applying ellipses include hard-coded width length for the label chips
FIX: changed the wrapper element for the v-chip components to preserve their default resizing behavior
Co-authored-by: Ali HAMDI <ahamdi@exoplatform.com>

Amine0008 committed:
undid changes in task47237 where we removed the users avatar for the activity  stream page (#1167)

ISSUE: reverting to previous behaviour where we removed the users avatar
FIX: undid changes in task 47237

abenmbarek committed:
TASK-51126 : User's profile: Spaces in common: incorrect values (#1155)

ISSUES : incorrect counting of common spaces values in the rest service
FIX : add a new SQL query that returns the common spaces between two given users and use it into a new added service.
without adding unit tests

Amine0008 committed:
task#47237: remove avatar inside comment block (#1164)

ISSUE: UX improvement, increasing the comment block size by removing the avatar image
FIX: removed the avatar image for comments and readjusted the margin styling

Amine0008 committed:
49308: Add Ellipsis for the blue chips on the settings page for responsive layout (#1129)

ISSUE: in the settings page we need to display the blue label chips with ellipsis for mobile web
FIX: added custom style that will be applied to these chips only when the application is opened on a mobile browser

houssemexo26 committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

Noussairhamrit committed:
Task-52783 : As a space manager, I'm unable to remove me as manager (#1191)

ISSUES : As a space manager, I'm unable to remove me as manager
FIX: To resolve this issue I added a condition in the method enabledProfileActionExtensions if user is logged in and a manager on a space the button remove Manager is displayed

Noussairhamrit committed:
Task-49046 : adding user have the same name as deleted one

ISSUES : when adding user have the same name as deleted one  an error message is displayed, with an unknown error message
FIX: To resolve this issue I added a condition in the method 'saveUser' if response status (500)  and response text 'Unable to create a previously deleted user : login' so a new error message displayed
	this error message is declared in 'Portlet_en.properties'

ghost commented:
### CodeSee Review Map:
[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/Meeds-io/social/1192/94c09049/5c94db8017764a208394857fde18acbcbcaf7321.svg)](https://app.codesee.io/r/reviews?pr=1192&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

Review [in an interactive map](https://app.codesee.io/r/reviews?pr=1192&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

View [more CodeSee Maps](https://app.codesee.io/r/maps?src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

ahamdi commented on webapp/portlet/src/main/webapp/vue-apps/idm-users-management/components/UsersManagementUserFormDrawer.vue:
```diff
@@ -236,6 +236,11 @@ export default {
             });
+          } else if (resp.status === 500){
+            return resp.text().then(error => {
+              this.fieldError = error;
+              throw new Error(this.$t('IDMManagement.error.LoginForDeletedUser'));
```
This will depend on the returned error, having error HTTP 500 does not mean that the username exists.
Besides checking the Rest service I found that there is a HTTP error 400 returned when the username is already found , could you recheck  what is the real cause of the server error?

Noussairhamrit committed:
Task-49046 : adding user have the same name as deleted one

ISSUES : when adding user have the same name as deleted one  an error message is displayed, with an unknown error message
FIX: To resolve this issue I worked on two  projects gatein-portal and social so in social  firstly i throw ObjectAlreadyExistsException instead of RuntimeException in 'SocialUserEventListenerImpl' also
I added a condition in the method 'handleError' if Error === 'USER:ALREADY_EXISTS_DELETED new error message displayed this error message is declared in 'Portlet_en.properties'

rdenarie commented on webapp/portlet/src/main/webapp/vue-apps/idm-users-management/components/UsersManagementUserFormDrawer.vue:
```diff
@@ -259,6 +259,8 @@ export default {
             const usernameError = this.fieldError.replace('USERNAME:', '');
             this.$refs.userNameInput.setCustomValidity(usernameError);
           }
+        } else if (this.fieldError === 'USER:ALREADY_EXISTS_DELETED') {
```
Move this in the previous if

rdenarie commented on webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties:
```diff
@@ -400,6 +400,7 @@ UsersManagement.title.deleteCurrentUserWarning=Can't delete user
 UsersManagement.message.deleteCurrentUserWarning=You can\u2019t delete your user account while being logged in with it.
 UsersManagement.message.userWithSameNameAlreadyExists=A user account with the same name already exists.
 UsersManagement.message.userWithSameEmailAlreadyExists=A user account with the same email already exists.
+UsersManagement.message.LoginForDeletedUser=The login already exists in a deleted account.
```
```suggestion
UsersManagement.message.LoginForDeletedUser=The username was already used for a previous deleted account. It cannot be reused.
```

Noussairhamrit committed:
code refactoring

Noussairhamrit committed:
refactoring code

Noussairhamrit committed:
code refactoring

rdenarie commented on component/api/src/main/java/org/exoplatform/social/common/ObjectAlreadyExistsException.java:
```diff
@@ -1,26 +1,18 @@
 package org.exoplatform.social.common;
```
Can you add also the copyright header ?
Thanks

rdenarie commented on component/core/src/main/java/org/exoplatform/social/core/listeners/SocialUserEventListenerImpl.java:
```diff
@@ -20,6 +20,7 @@
 import java.util.Arrays;
```
Copyright header

Noussairhamrit committed:
add copyright header

rdenarie commented:
Please be sure to merge the gatein-portal PR before restarting the build of the job for this PR

rdenarie reviewed (changes requested):
Can you fix the code smells related to sonar (Except "Do not forget to remove this deprecated code someday."

Amine0008 committed:
task45533: added a settings page where the user can download the mobile application


added a settings page where the user can download the mobile application by either scanning a QR code or navigating directly to the relevant app store by clicking on the appropriate store icon

Noussairhamrit committed:
fix sonar Code Smells

Noussairhamrit committed:
fix sonar Code Smells

rdenarie commented on component/api/src/main/java/org/exoplatform/social/common/ObjectAlreadyExistsException.java:
```diff
@@ -1,26 +1,38 @@
-  }
-
+/**
+ * @since 6.3.x
+ * @deprecated Deprecated from 6.3.x. Replaced by a new Exception org.exoplatform.commons.ObjectAlreadyExistsException
+ *             will be removed at 6.3.x
+ */
+@Deprecated(since="6.3.x", forRemoval=true)
```
```suggestion
 * @since 6.3.0
 * @deprecated Deprecated from 6.3.0. Replaced by a new Exception org.exoplatform.commons.ObjectAlreadyExistsException
 *             will be removed at 6.3.0
 */
@Deprecated(since="6.3.0", forRemoval=true)
```

Noussairhamrit committed:
code refactoring

houssemexo26 committed:
Merge branch 'feature/maintenance' into fix/49046

Noussairhamrit committed:
Merge branch 'feature/maintenance' into fix/49046
